src/spatial_reaction_systems/utility.jl

```diff
@@ -0,0 +1,298 @@
+    merge(vals_to_dict(vertex_parameters(lrs), vert_ps),
+          vals_to_dict(edge_parameters(lrs), edge_ps))
+end
+
+# Computes the transport rates and stores them in a desired format 
+# (a Dictionary from species index to rates across all edges).
+function compute_all_transport_rates(vert_ps::Vector{Vector{Float64}}, edge_ps::Vector{Vector{Float64}}, lrs::LatticeReactionSystem)
+    # Creates a dict, allowing us to access the values of wll parameters.
+    p_val_dict = param_dict(vert_ps, edge_ps, lrs)    
+
+    # For all species with transportation, compute their transportation rate (across all edges). 
+    # This is a vector, pairing each species to these rates.
+    unsorted_rates = [s => compute_transport_rates(get_transport_rate_law(s, lrs), p_val_dict, lrs.num_edges) 
+                        for s in spatial_species(lrs)] 
+    
+    # Sorts all the species => rate pairs according to their species index in species(::ReactionSystem).
+    return sort(unsorted_rates; by=rate -> findfirst(isequal(rate[1]), species(lrs)))   
+end
+# For a species, retrieves the symbolic expression for its transportation rate
+# (likely only a single parameter, such as `D`, but could be e.g. L*D, where L and D are parameters).
+# We could allows several transportation reactions for one species and simply sum them though, easy change.
+function get_transport_rate_law(s::BasicSymbolic{Real}, lrs::LatticeReactionSystem)
+    rates = filter(sr -> isequal(s, sr.species), lrs.spatial_reactions)
+    (length(rates) > 1) && error("Species $s have more than one diffusion reaction.")    
+    return rates[1].rate
+end
+# For the numeric expression describing the rate of transport (likely only a single parameter, e.g. `D`), 
+# and the values of all our parameters, computes the transport rate(s).
+# If all parameters the rate depend on are uniform all edges, this becomes a length 1 vector.
+# Else a vector with each value corresponding to the rate at one specific edge.
+function compute_transport_rates(rate_law::Num,
+                                p_val_dict::Dict{SymbolicUtils.BasicSymbolic{Real}, Vector{Float64}}, num_edges::Int64)
+    # Finds parameters involved in rate and create a function evaluating teh rate law.
+    relevant_ps = Symbolics.get_variables(rate_law)
+    rate_law_func = drop_expr(@RuntimeGeneratedFunction(build_function(rate_law, relevant_ps...)))
+
+    # If all these parameters are spatially uniform. `rates` becomes a vector with 1 value.
+    if all(length(p_val_dict[P]) == 1 for P in relevant_ps)  
+        rates = [rate_law_func([p_val_dict[p][1] for p in relevant_ps]...)]
+    # If at least on parameter the rate depends on have a value varying across all edges,
+    # we have to compute one rate value for each edge.
+    else
+        rates = [rate_law_func([get_component_value(p_val_dict[p], idxE) for p in relevant_ps]...) 
+                    for idxE in 1:num_edges]
+    end
+    return Symbolics.value.(rates)
```

Comment:
Why is this needed? I thought `rate_law_func` should now return a numeric value?

Reply:
you are right, this shouldn't be required anymore.

Reply:
OK, once that and compat are fixed feel free to merge. 

Awesome job on this @TorkelE thank you for your patience and endurance!!!